jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostModel.scala

```diff
@@ -89,7 +89,7 @@ class XGBoostModel(_booster: Booster)(implicit val sc: SparkContext) extends Ser
    *
    * @param modelPath The model path as in Hadoop path.
    */
-  def saveModelAsHadoopFile(modelPath: String): Unit = {
+  def saveModelAsHadoopFile(modelPath: String)(implicit sc: SparkContext): Unit = {
     val path = new Path(modelPath)
     val outputStream = path.getFileSystem(sc.hadoopConfiguration).create(path)
```

Comment:
the modification seems reasonable...would you make it pass the unit test so that I can merge?